src/main/java/org/spongepowered/common/command/CommandPermissions.java

```diff
@@ -53,15 +54,28 @@ public static boolean testPermission(CommandSource source, String commandName) {
         if (commandName.equals(CommandPermissions.COMMAND_BLOCK_COMMAND)) {
             return source.hasPermission(CommandPermissions.COMMAND_BLOCK_PERMISSION);
         }
-        Optional<? extends CommandMapping> mapping = SpongeImpl.getGame().getCommandManager().get(commandName);
-        if (mapping.isPresent()) {
-            CommandCallable callable = mapping.get().getCallable();
-            if (callable instanceof MinecraftCommandWrapper) {
-                return source.hasPermission(((MinecraftCommandWrapper) callable).getCommandPermission());
-            } else {
-                return callable.testPermission(source);
-            }
+        //This method is only called for Minecraft/Modded commands, so don't look at commands provided by plugins.
+        final List<? extends CommandMapping> nativeMappings = SpongeImpl.getGame().getCommandManager().getAll(commandName).stream()
+                .filter(m -> m.getCallable() instanceof MinecraftCommandWrapper)
+                .collect(Collectors.toList());
+        //Forge rules for clashing commands is whatever registers last wins, but secondary aliases never override primary.
+        //thus we never expect there to be more then one native mapping for a given alias.
+        //Possible I guess, if some plugin (not mod) intentionally re-registers a minecraft/mod command under a clashing alias...
```

Comment:
Hey JBYoshi, will probably get around to testing this, this afternoon.

But just wanted to ask if you had tested this? From what I could see (without testing purposely clashing forge commands) is that forge will overwrite the tp mapping with SpongePermissionTestForge's permission mapping.

So unless this is covered by a mixin somewhere it should be fine?

Anyway I'll try to get around to compiling this test-mod at some stage to have hard imperative proof one way or another.